src/Tribe/Events/Attendee_List_Display.php

```diff
@@ -0,0 +1,231 @@
+		 *
+		 * @return bool
+		 */
+		$is_showing_attendee_list = (bool) apply_filters(
+			'tribe_tickets_event_is_showing_attendee_list',
+			$is_showing_attendee_list,
+			$post,
+			$this->is_using_blocks()
+		);
+
+		update_post_meta( $post->ID, Tribe__Tickets_Plus__Attendees_List::HIDE_META_KEY, $is_showing_attendee_list );
+	}
+
+	/**
+	 * This keeps track of whether the Attendee List is being displayed becase of a shortcode
+	 * in the content, and acts accordingly if said shortcode is removed.
+	 *
+	 * @param WP_Post $post The Post being checked.
+	 *
+	 * @return void
+	 */
+	private function track_shortcode_driven_meta( WP_Post $post ) {
+		$is_visible_by_meta                  = Tribe__Tickets_Plus__Attendees_List::is_hidden_on( $post ) === false;
+		$has_attendee_list_shortcode         = has_shortcode( $post->post_content, 'tribe_attendees_list' );
+		$has_attendee_list_by_shortcode_meta = get_post_meta( $post->ID, self::$attendee_list_by_shortcode, true ) === 'yes';
+
+		/*
+		 * If what triggers the Attendee List to display is a shortcode in the content,
+		 * let's save this piece of information so that we revert it when the shortcode is removed.
+		 */
+		if ( ! $is_visible_by_meta && $has_attendee_list_shortcode ) {
+			update_post_meta( $post->ID, self::$attendee_list_by_shortcode, 'yes' );
+		}
+
+		/*
+		 * The shortcode that triggered the Attendee List to display is no longer in the content.
+		 * Thus, we shall update the "Attendee List" meta to "hidden" again.
+		 */
+		if ( $has_attendee_list_by_shortcode_meta && ! $has_attendee_list_shortcode ) {
+			update_post_meta( $post->ID, self::$attendee_list_by_shortcode, 'no' );
+
+			add_filter( 'tribe_tickets_event_is_showing_attendee_list', '__return_false' );
+		}
+	}
+
+	/**
+	 * @return bool
+	 */
+	private function is_using_blocks() {
+		$should_load_blocks          = tribe( 'editor' )->should_load_blocks();
+		$is_blocks_active_for_events = tribe_is_truthy( tribe_get_option( 'toggle_blocks_editor', false ) );
+		$is_using_blocks             = $should_load_blocks && $is_blocks_active_for_events;
+
+		/**
+		 * Whether post content is being served through blocks
+		 * or the classical editor.
+		 *
+		 * @since TBD
+		 *
+		 * @param bool $is_using_blocks True if using blocks. False if using the classical editor.
+		 */
+		$is_using_blocks = (bool) apply_filters( 'tribe_is_using_blocks', $is_using_blocks );
+
+		return $is_using_blocks;
+	}
+
+	/**
+	 * @param WP_Post $post The Post being checked.
+	 *
+	 * @return bool
+	 */
+	private function is_showing_attendee_list_with_blocks( WP_Post $post ) {
+		$has_attendee_list_block     = has_block( 'tribe/attendees', $post );
+		$has_attendee_list_shortcode = has_shortcode( $post->post_content, 'tribe_attendees_list' );
+
+		return $has_attendee_list_block || $has_attendee_list_shortcode;
+	}
+
+	/**
+	 * @param WP_Post $post The Post being checked.
```

Comment:
```suggestion
	 * Determine whether we are showing the attendee list with the block editor.
	 *
	 * @since TBD
	 *
	 * @param WP_Post $post The Post being checked.
```